src/app/components/shared/ConfirmScaffoldForm/ConfirmScaffoldForm.tsx

```diff
@@ -0,0 +1,54 @@
+import { Spinner } from "@amsterdam/asc-ui"
+import { ScaffoldForm } from "@amsterdam/amsterdam-react-final-form"
+import ScaffoldFields, { Fields } from "app/components/shared/Form/ScaffoldFields"
+import ConfirmScaffoldFields from "./ConfirmScaffoldFields"
+import useSubmitConfirmation from "./hooks/useSubmitConfirmation"
+
+type Props<T, U> = {
+  fields: { fields: Fields } | undefined
+  postMethod: (data: T) => Promise<U>
+  afterSubmit?: (result: U) => Promise<unknown>
+  initialValues?: Record<string, unknown>
+}
+
+const ConfirmScaffoldForm = <T extends Record<string, any>, U extends Record<string, any>>(props: Props<T, U>) => {
```

Comment:
What's happening here?

Reply:
That allows for generic React components. But will check if it's possible to make it more readable.